### PR TITLE
NRF crash issue fix

### PR DESCRIPTION
**Pull request by vthiruveedula**

### CODE REVIEW

File: dbadapter/dbadapter.go [hunk 1]
```diff
@@ -49,10 +49,16 @@ func iterateChangeStream(routineCtx context.Context, stream *mongo.ChangeStream)
...
 	dbc := &MongoDBClient{}
-	MongoDBLibrary.SetMongoDB(setdbName, url)
-	if MongoDBLibrary.Client != nil {
-		dbc.Client = MongoDBLibrary.Client
-		dbc.dbName = setdbName
+	for {
+		MongoDBLibrary.SetMongoDB(setdbName, url)
+		if MongoDBLibrary.Client != nil {
+			dbc.Client = MongoDBLibrary.Client
+			dbc.dbName = setdbName
```
Comment (thakurajayL) [inline dbadapter/dbadapter.go:56]: Use new DB library now.

File: producer/nf_management.go [hunk 2]
```diff
@@ -284,6 +286,19 @@ func NFDeregisterProcedure(nfInstanceID string) (problemDetails *models.ProblemD
...
 
+func sendNFDownNotification(nfProfile models.NfProfile, nfInstanceID string) {
+	if nfProfile.NfType == models.NfType_AMF {
+		url := "http://amf:29518" + "/namf-oam/v1/amfInstanceDown/" + nfInstanceID
+		req, err := http.NewRequest(http.MethodPost, url, nil)
+		req.Header.Set("Content-Type", "application/json")
+		client := &http.Client{}
+		_, err = client.Do(req)
+		if err != nil {
+			fmt.Println("Errored when sending request to the server")
```
Comment (badhrinathpa) [inline producer/nf_management.go:297]: lets print the error here

File: go.mod [hunk 3]
```diff
@@ -2,14 +2,16 @@ module github.com/omec-project/nrf
 
 go 1.16
 
+replace github.com/omec-project/nrf => ./
```
Comment (badhrinathpa) [inline go.mod:5]: we can remove this before merging right ?

File: producer/nf_management.go [hunk 4]
```diff
@@ -284,6 +286,19 @@ func NFDeregisterProcedure(nfInstanceID string) (problemDetails *models.ProblemD
 	return nil
 }
 
+func sendNFDownNotification(nfProfile models.NfProfile, nfInstanceID string) {
```
Comment (badhrinathpa) [inline producer/nf_management.go:289]: Is this part of the spec notification model or a different API?

**Review by badhrinathpa — APPROVED** [review]
+1

**Pull request merged**